### PR TITLE
d2a2c_vect is an object

### PR DESCRIPTION
**Pull request by elynnwu**
## Purpose

OOPify `d2a2c_vect`

## Code changes:

- `DGrid2AGrid2CGridVectors` is a new class in `d2a2c_vect`
- Change calls to `d2a2c_vect` in `c_sw`

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--BD7zigBMAhMZAPkeNENeuU2UAg-IlsYffZgTwyKEylty7NhY).
- [x] Docstrings and type hints are added to new and updated routines, as appropriate
- [x] Unit tests are added or updated for non-stencil code changes

### CODE REVIEW

File: fv3core/stencils/d2a2c_vect.py [hunk 1]
```diff
@@ -380,187 +363,291 @@ def edge_interpolate4_y(va, dya):
...
+            domain=(self.grid.nic + pad, self.grid.njc + pad, self.grid.npz),
+        )
+        origin_edges = self.grid.compute_origin(add=(-3, -3, 0))
+        domain_edges = self.grid.domain_shape_compute(add=(6, 6, 0))
+        ax_offsets_edges = axis_offsets(self.grid, origin_edges, domain_edges)
+        self._fill_corners_x = FrozenStencil(
+            func=fill_corners_x,
+            externals={
+                **ax_offsets_edges,
+            },
```
Comment (jdahm) [inline fv3core/stencils/d2a2c_vect.py:471]:
```suggestion
            externals=ax_offsets_edges,
```

File: fv3core/stencils/d2a2c_vect.py [hunk 2]
```diff
@@ -380,187 +363,291 @@ def edge_interpolate4_y(va, dya):
...
+
+        self._vt_main = FrozenStencil(
+            func=vt_main,
+            origin=(i1, jfirst, 0),
+            domain=(self.grid.nic + 2, jdiff, self.grid.npz),
+        )
+
+    def __call__(self, uc, vc, u, v, ua, va, utc, vtc):
+        """
+        Calculate velocity vector from D-grid to A-grid to C-grid
```
Comment (jdahm) [inline fv3core/stencils/d2a2c_vect.py:526]:
```suggestion
        Calculate velocity vector from D-grid to A-grid to C-grid.
```

File: fv3core/stencils/c_sw.py [hunk 3]
```diff
@@ -508,7 +512,8 @@ def __call__(
             self.delpc,
             self.ptc,
         )
-        d2a2c.compute(self._dord4, uc, vc, u, v, ua, va, ut, vt)
+        # d2a2c.compute(self._dord4, uc, vc, u, v, ua, va, ut, vt)
```
Comment (rheacangeo) [inline fv3core/stencils/c_sw.py:515]: can remove the commented code

File: fv3core/stencils/d2a2c_vect.py [hunk 4]
```diff
@@ -3,9 +3,10 @@
 
 import fv3core._config as spec
 import fv3core.utils.gt4py_utils as utils
-from fv3core.decorators import gtstencil
+from fv3core.decorators import FrozenStencil
```
Comment (rheacangeo) [inline fv3core/stencils/d2a2c_vect.py:6]: I like to see this!!

File: tests/translate/translate_d2a2c_vect.py [hunk 5]
```diff
@@ -16,7 +18,7 @@ def __init__(self, grid):
             "utc": {},
             "vtc": {},
         }
-        self.in_vars["parameters"] = ["dord4"]
+        # self.in_vars["parameters"] = ["dord4"]
```
Comment (rheacangeo) [inline tests/translate/translate_d2a2c_vect.py:21]: can delete

**Review by rheacangeo — APPROVED** [review]
Awesome!! I approve, just a couple of minor cleanup things, and Johann's suggestions make sense to do.

File: fv3core/stencils/d2a2c_vect.py [hunk 6]
```diff
@@ -380,187 +363,291 @@ def edge_interpolate4_y(va, dya):
...
+                self._utmp,
+                self._vtmp,
+            )
+        if self.grid.east_edge:
+            self._avg_box_e(
+                u,
+                v,
+                self._utmp,
+                self._vtmp,
+            )
```
[the remark this answers is not in the document]
Comment (elynnwu) [reply fv3core/stencils/d2a2c_vect.py]: Great suggestion, thanks for the tips!

**Pull request merged**